lib/ansible/plugins/callback/default.py

```diff
@@ -190,7 +190,7 @@ def _print_task_banner(self, task):
 
         # Use cached task name
         task_name = self._last_task_name
-        if task_name is None:
+        if task_name is None or self._play.strategy == 'free':
```

Comment:
I realize there is already another special case for `free`, but it makes me wonder if there's a way we can handle this difference in behavior without having to include a special case for the `free` strategy. Perhaps an attribute on strategies indicating if they need cached task names?

Reply:
Perhaps, but that's not something I want to try to solve in this PR.

Reply:
Understood. I just wanted to mention it since this didn't look quite right. I'm OK with this for the immediate fix.